### PR TITLE
PC-style numpad language pack

### PR DESCRIPTION
**Pull request by MrSmoer**
Dear AnySoftKeyboard maintainers,
I needed a PC-style numpad keyboard on my phone to play a muscle-memory-number-remembering-game I wrote in python, so I took the standard Phone-style numpad and exchanged the first and the third row.

I've got to say, your contributing documentation was very helpful.

Potential Issues
- [ ] Unfortunately, I could not figure out, if it is possible to switch between two different variants of a keyboard, depending on the resolution, as the standard Phone-Style numpad does.
- [ ] The flag.svg is very large ~3.5 MB
- [ ] It is not set to publish yet
- [x] I think the screenshots are still wrong
- [ ] I did not know which locale I had to choose, so I chose `np`
- [ ]  There is warning before commiting `Warning:(5, 10) Exported receiver does not require permission` from `addons/languages/numpad/pack/src/main/AndroidManifest.xml`

Feel free to contact me for the smallest issue.

Have a nice day!

### CODE REVIEW

**Review by menny — COMMENTED** [review]
The SVG is pretty large, can you find something smaller? Not a biggy, though

File: addons/languages/numpad/pack/build.gradle [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//ext.dictionaryInputAdditionalInnerCharacters = null//"".toCharArray()
+ext.shouldGenerateDictionary = false
+
+apply from: "$rootDir/addons/gradle/language_pack_lib.gradle"
+//apply from: "$rootDir/gradle/publish_maven.gradle"
+//
+//publish {
+//    artifactId = 'english'
+//    desc = "English pack for AnySoftKeyboard"
+//}
```
Comment (menny) [inline addons/languages/numpad/pack/build.gradle:13]: this is confusing, can you remove this

File: addons/languages/numpad/pack/src/main/res/values/numpad_pack_strings_dont_translate.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources xmlns:tools="http://schemas.android.com/tools"
+           xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2"
+           tools:ignore="MissingTranslation,TypographyDashes">
+
+    <string name="numpad_keyboard">Numpad</string>
+   <!-- <string name="main_english_keyboard_id">c7535083-4fe6-49dc-81aa-c5438a1a343a</string>-->
```
Comment (menny) [inline addons/languages/numpad/pack/src/main/res/values/numpad_pack_strings_dont_translate.xml:7]: this should also go away

**Comment by MrSmoer** [comment]
> The SVG is pretty large, can you find something smaller? Not a biggy, though

I think I would like to leave it that way.

File: addons/languages/numpad/pack/src/main/res/values/numpad_pack_strings_dont_translate.xml (hunk 2 above)
Comment (MrSmoer) [reply addons/languages/numpad/pack/src/main/res/values/numpad_pack_strings_dont_translate.xml]: Just the comment, or the whole file? 

The comment is removed now

File: addons/languages/numpad/pack/build.gradle (hunk 1 above)
Comment (MrSmoer) [reply addons/languages/numpad/pack/build.gradle]: Done

**Comment by MrSmoer** [comment]
There is a warning when building from `addons/languages/numpad/pack/src/main/AndroidManifest.xml`, it says that `Warning:(5, 10) Exported receiver does not require permission`, is that fine?

**Comment by MrSmoer** [comment]
I had accidentally removed the entry for the ic_status_numpad.png file therefore the check failed, it is fixed now

**Comment by MrSmoer** [comment]
I have tested your workflow on my fork and it passed till after the all green requirement  so it should be fine now

**Review by menny — APPROVED** [review]

**Comment by MrSmoer** [comment]
I've had quite a few for this pack, could you maybe publish this on the PlayStore and/or F-Droid?
In the Contributing guide it says "the pack needs to be owned by the AnySoftKeyboard developer account", I don't know how to go about this, but I guess its in case I would have already published it.
Thank you very much!

**Comment by menny** [comment]
It is now available on the Play Store: [View on Google Play](https://play.google.com/store/apps/details?id=com.anysoftkeyboard.languagepack.numpad)

**Comment by therealmaxmoon** [comment]
>  I needed a PC-style numpad keyboard on my phone to play a muscle-memory-number-remembering-game I wrote in python, so I took the standard Phone-style numpad and exchanged the first and the third row.

May I ask if you published the muscle memory game and how exactly did you manage to get a pc numpad on ASK?

I am playing [this app](https://f-droid.org/en/packages/io.github.printn.humanbenchmark/) currently and it has a muscle memory game, too, but I am annoyed of the upside down numpad and if I activate [this pc numpad plug in](https://f-droid.org/en/packages/com.anysoftkeyboard.languagepack.numpad/), I can't use my keyboard for typing anymore, because all letters are gone.

**Comment by MrSmoer** [comment]
> >  I needed a PC-style numpad keyboard on my phone to play a muscle-memory-number-remembering-game I wrote in python, so I took the standard Phone-style numpad and exchanged the first and the third row.
> 
> May I ask if you published the muscle memory game and how exactly did you manage to get a pc numpad on ASK?
> 
> I am playing [this app](https://f-droid.org/en/packages/io.github.printn.humanbenchmark/) currently and it has a muscle memory game, too, but I am annoyed of the upside down numpad and if I activate [this pc numpad plug in](https://f-droid.org/en/packages/com.anysoftkeyboard.languagepack.numpad/), I can't use my keyboard for typing anymore, because all letters are gone.

Right now I just installed ASK and the Addon and it just worked out of the Box, I am not sure what  you mean by that.

Can't you just use this button to switch back to your main keyboard?


My number remembering game works within Termux and is accessible on my github, its called [pi-trainer](https://github.com/MrSmoer/pi-trainer)

**Comment by therealmaxmoon** [comment]
> Right now I just installed ASK and the Addon and it just worked out of the Box, I am not sure what you mean by that.

Ohhhh, I've got upper row configured as a number row, that's why the language button wasn't there and it isn't possible to cycle through it with the space bar. After deactivating the upper number row it works again. Thanks :)

But still, the app I mentioned above doesn't use the pc-numpad. The first row still has 1-2-3. If I switch to a messenger and try to type number, the first row is 7-8-9. That's really weird. 

> My number remembering game works within Termux and is accessible on my github, its called [pi-trainer](https://github.com/MrSmoer/pi-trainer)

I was wondering how you would execute a python script on Android. I tried to execute it with Termux like you wrote it in the README, but I wasn't able to type anything, way to complex. But it's a pretty cool game for Linux, I like the Senso mode.

**Comment by MrSmoer** [comment]
> But still, the app I mentioned above doesn't use the pc-numpad. The first row still has 1-2-3. If I switch to a messenger and try to type number, the first row is 7-8-9. That's really weird. 

That is probably because of them having InputType.TYPE_CLASS_NUMBER, which summons a different view I believe. I would say this is a bug in the extension, I will take a look at that.

> I was wondering how you would execute a python script on Android. I tried to execute it with Termux like you wrote it in the README, but I wasn't able to type anything, way to complex. But it's a pretty cool game for Linux, I like the Senso mode.

You just need to install python and git within termux, clone the pi-trainer repo inside termux, and run it by navigating to the repo with cd and run the sh file. Maybe open an issue on the repo, it is kind of off-topic here 🙃

**Pull request merged**